contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -493,7 +494,7 @@ Datum remove_accents_internal(PG_FUNCTION_ARGS)
 
 		// Open transliterator
 		cached_transliterator = utrans_openU(rules, len_uchar, UTRANS_FORWARD, NULL, 0, NULL, &status);
-		if (U_FAILURE(status))
+		if (U_FAILURE(status) || !cached_transliterator)
 		{
 			ereport(ERROR,
```

Comment:
Should we free `rules` before error, since it is in topmemory context

Reply:
we want to reuse rules, that's why we are caching it in top memory context

Reply:
Good catch! But we dont need to free it as it would not have initialised in case of error.

Reply:
[Doc](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/utrans_8h.html#a2fc501ae07ce3ce56b164db8636e9f12) of utrans_openU says 
```
Returns
    a transliterator pointer that may be passed to other utrans_xxx() functions, or NULL if the open call fails.
````